docs/grammar.md

```diff
@@ -0,0 +1,118 @@
+# RON grammar
+
+This file describes the structure of a RON file in [EBNF notation][ebnf].
+If extensions are enabled, some rules will be replaced. For that, see the
+[extensions document][exts] which describes all extensions and what they override.
+
+[ebnf]: https://en.wikipedia.org/wiki/Extended_Backus–Naur_form
+[exts]: ./extensions.md
+
+## RON file
+
+```ebnf
+RON = [extensions], ws, value, ws
+```
+
+## Whitespace and comments
+
+```ebnf
+ws = { ws_single, comment }
+ws_single = "\n" | "\t" | "\r" | " "
+comment = ["//", { no_newline }, "\n"]
+```
+
+## Commas
+
+```ebnf
+comma = ws, ",", ws
+```
+
+## Extensions
+
+```ebnf
+extensions = { "#", ws, "!", ws, "[", ws, extensions_inner, ws, "]", ws }
+extensions_inner = "enable", ws, "(", extension_name, { comma, extension_name }, [comma], ws, ")" 
+```
+
+For the extension names see the [`extensions.md`][exts] document.
+
+## Value
+
+```ebnf
+value = unsigned | signed | float | string | char | bool | option | list | map | tuple | struct | enum_variant
+```
+
+## Numbers
+
+```ebnf
+digit = "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+unsigned = digit, { digit }
```

Comment:
we've got to support other numerical systems (I know we don't support them atm)

Reply:
Agreed